### ISSUE TITLE
4 to 8 directional movement for characters.

### ISSUE THREAD

**Issue by WithinAmnesia** [jonathaneeckhout/jdungeon#243]
4 to 8 directional movement for characters.
https://raw.githubusercontent.com/WithinAmnesia/ARPG/main/C.C.0.%20Assets%20Archive/Art/FoozleCC/Lucifer/Foozle_2DC0009_Lucifer_Warrior_Pixel_Art/Warrior%20Showreel.gif
https://github.com/jonathaneeckhout/jdungeon/issues/241
Most 2D RPGs have 4 or 8 directional movement. Many simple single player Godot 4 RPGs have at least 4 directional movement and it looks cool. How do we do this?

**Comment by jonathaneeckhout** [comment]
This can definitely be done later in a later phase of the project. Currently the game only has 2 face direction for sprites (and actually just 1 as it just flips the sprites) because of simplicity. Artists only need to draw a sprite once and it will fit the game. Simplicity is one of the key features for this project as it allows us to attract new people to contribute towards the project.

**Comment by WithinAmnesia** [comment]
That is good to hear; I hope the timing is fairly soon after the web support for this 4 to 8 directional ,movement feature affects the majority of the game experience quality reasonably possible with this MMORPG.

I had troubles with using the simple art format for it was an unusual art format unlike most community standard 4 directional movement sprite sheets. I literally downloaded every C.C.0 RPG art work from itch.io and open game art after many hours to build up a C.C.0 creative community art asset library and archive for every public 2D 2.5D and 3D community C.C.0 assets.

I can say after many hours of collecting and archiving C.C.0 creative community sprite art assets that this art format is very uncommon if not borderline unused in either dead flash games or prototypes. For every art format creative community C.C.0 art asset that matches this strange art format you will find 10+ 4  directional 16x16 to 32x32 to 64x64 pixel dimensions sprite and tile creative community C.C.0 art assets.

I had to scrap lots of great C.C.0 work from some of the best community animators to fit to the unusual head, chest, 2 arms and 2 legs format. It was discouraging to butcher masterwork pixel animation to fit a very niche and weird art format and know I have literally thousands of C.C.0 art assets that must be butchered to fit this 'simple', 'easy' and approachable art format that is rarely encountered in the creative community.

I have worked on many different games from 2D, 2.5D, 3D from the 80's, 90's, 2000's, 2010's and 2020's very few dare use anything other than 4 to 8+ direction movement. Unless it is art work that is not an animated character sprite; let alone a player character sprite.

At first this 5 floating body segment pieces art format looks simple yet it is not future proof and creatively limiting. It would objectively be better and more future proof and more in line with the creative community's mountains and decades of C.C.0 art work and values to use proper art animation formats. Such as the 4 to 8 direction a movement art formats.

These proper art formats of 4 to 8 directional movement also support more room for more diverse dimension animations such as boss battles and large and small characters. As while using the same traditional 2D and 2.5D animation systems of sequential sprite frames that can be mixed with rotating sprite frames akin to 3D animation.

Proper 4 and 8 directional movement can be compatible to most proper 2D and 2.5D art of the creative community and have less stick figure animations limitations and no development dead ends for an MMORPG style game that hopes to launch in the 2020's.

In 1986 the Legend of Zelda has 4 directional movement for characters and there are many free Godot 4 tutorial video series on sites like Youtube for 2D and 2.5D 4 directorial movement and with proper use / attack animations.

As with 95+% of an ARPG is viewing the player sprite. This the character and most definitely the player character animation format should be of high quality. Especially with animations to give a good start for the first seconds of players first impressions in the critical ~40 second first play window. As well as to build future proofing and creative depth for other non-player characters; such as mobs, encounters and civilians etc. within game worlds.

As with 4 to 8 directional character movement systems are what people expect 2D and 2.5D RPGs ARPGs and MMORPGs to have as standard features. For 4 to 8 directional character movement systems are the core animation foundation for such classic ~RPG series as Ultima Online, Classic RuneScape, Zelda, Final Fantasy, Diablo, Baldur's Gate, Fallout, Warcraft, Mario, Age of Empires etc. and the list goes on from the 70' to and into the future decades.

Proper 2D and 2.5D animation formats help creative efficiency and world building and boost player experience starting from a excellent player character animations and movement at the start of games all the way to future demand of more adventure and gameplay expansions worth playing.
